docs/advanced/pss.md

```diff
@@ -0,0 +1,166 @@
+---
+title: PSS Messaging
+id: pss
+---
+
+Out of the ashes of Ethereum's vision for a leak-proof decentralised anonymous messaging system - Whisper - comes PSS. Swarm provides the ability to send messages that appear to be normal Swarm traffic, but are in fact messages that may be received and decrypted to reveal their content only by the specific nodes they were intended to be received by.
+
+PSS provides a pub-sub facility that can be used for a variety of tasks. Nodes are able to listen to messages received for a specific topic in their nearest neighbourhood and create messages destined for another neighbourhood which are sent over the network using Swarm's usual data dissemination protocols.
+
+### Enable PSS On Your Node
+
+In order to receive PSS messages you must first enable the p2p websockets API when starting your node.
+
+```sh
+	bee start --p2p-ws-enable
+```
+
+### Subscribe and Receive Messages
+
+Once your Bee node is up and running, you will be able to subscribe to feeds using websockets. For testing, it is useful to use the [websocat](https://docs.rs/crate/websocat/1.0.1) command line utility.
+
+Here we subscribe to the topic `test-topic`
+
+```sh
+websocat ws://localhost:8080/pss/subscribe/test-topic
+```
+
+Our node is now watching for new messages received in it's nearest neighbourhood.
+
+### Send Messages
+
+Messages can be sent simply by sending a `POST` request to the PSS api endpoint.
+
+When sending messages, we must specify a 'target' prefix of the recipients swarm address, a partial address representing their neighbourhood. Currently the length of this prefix is recommended to be two bytes, which will work well until the network has grown to a size of ca. 20-50K nodes. 
+
+For example, if we want to send a PSS message to a node with address `49d80b28adccd67a09959a601f0e28521aaeb06bf1bf17e9e012e7b597c3d926`, we must include the 'target' `49d8`.
+
+```sh
+curl -XPOST localhost:8082/pss/send/test/49d8 --data "Hello Swarm"
+```
+
+### Send Messages in a Test Network
+
+Now, let's see this in action by setting up two Bee nodes, connecting them, and sending PSS messages from one to the other.
+
+First start two Bee nodes. We will start them with distinct ports for the api, debug api and p2p port so that there are no conflicts, since they will be running on the same computer. 
+
+Run the following command to start the first node. Note that we are passing `""` to the `--bootnode` argument so that our nodes will not connect to the network.
+
+```sh
+bee start \
```

Comment:
If you're running a local two node network you should probably run with `--swap-enable=false`